docs/user-guide/coordinate-transformations.ipynb

```diff
@@ -427,11 +428,70 @@
    "cell_type": "code",
    "execution_count": null,
    "id": "15cbd635-be46-44d8-a125-30ec8b97d1cd",
-   "metadata": {},
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
    "outputs": [],
    "source": [
     "binned.plot(resolution={'y': 36, 'x': 24})"
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4857576d",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Renaming of Dimensions\n",
+    "\n",
+    "This section is somewhat advanced and not required to understand the basic usage of `transform_coords.`\n",
+    "\n",
+    "As shown above, `transform_coords` can rename dimensions of the data array if it processes dimension-coordinates.\n",
+    "This section describes in detail the rules how it is determined which dimensions are renamed.\n",
+    "The rules controlling this behavior are generic and tend to favor not renaming a dimension if it is not entirely clear what the new name should be.\n",
+    "\n",
+    "A dimension is only renamed if one output of `transform_coords` can be uniquely associated with the corresponding dimension-coordinate.\n",
+    "In the example above, there are two dimension-coordinates in the input, `datetime` and `location`.\n",
+    "The latter is used to compute two new coordinates, namely `longitude` and `latitude`.\n",
+    "It is thus not possible to find a unique new name for the `location` *dimension*.\n",
+    "`datetime` on the other hand is the only dimension-coordinate used to compute `local_datetime` and finally `local_time` (remember that `longitude` is not a dimension-coordinate).\n",
+    "The `datetime` *dimension* is therefore renamed to `local_time`.\n",
+    "\n",
+    "Identifying outputs with dimension-coordinates can be expressed as a graph-coloring problem.\n",
+    "We assign a unique color to each dimension-coordinate; other coordinates are left uncolored (black).\n",
+    "We then propagate colors through the directed graph using the following rules\n",
+    "\n",
+    "1. If a node has exactly one colored parent, use that parent's color (left).\n",
+    "2. If a node has several colored parents, leave the node black (middle).\n",
+    "   The same happens if no parent is colored.\n",
+    "3. If a node has more than one child, all the children are black (2nd from right).\n",
+    "   This is even the case when there is another colored node connected to a child (right).\n",
+    "\n",
+    "![base rules](../images/transform_coords/base_rules.svg)\n",
+    "\n",
+    "All graphs used by `transform_coords` are directed acyclic graphs, but they can still have undirected cycles.\n",
+    "In those cases, it can be possible to identify an output with a dimension-coordinate even when it has multiple children as part of a cycle.\n",
+    "\n",
+    "4. If a cycle has one and only one colored node as input (w.r.t. directions of the edges) and exactly one output, that output takes the color of the input.\n",
```

Comment:
Note that markdown renders the 4. as a 1.

Reply:
It doesn't for me. Both Jupyter Lab and sphinx docs show 4.

Reply:
Ok, may be a github thing.